### PR TITLE
Handle misc error codes todos. [DPP-606]

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala [hunk 1]
```diff
@@ -20,6 +19,18 @@ trait ContextualizedErrorLogger {
   def warn(message: String, throwable: Throwable): Unit
   def error(message: String): Unit
   def error(message: String, throwable: Throwable): Unit
+
+  /** Formats the context as a string for logging */
+  protected[error] def formatContextAsString(contextMap: Map[String, String]): String = {
```
Comment (pbatko-da) [inline ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala:24]: Moved from ErrorCode

Comment (skisel-da) [reply ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala:24]: I would move it to static object, as this has nothing to do with an instance

File: ledger/error/src/test/suite/scala/com/daml/error/DamlContextualizedErrorLoggerSpec.scala [hunk 2]
```diff
@@ -13,7 +13,8 @@ class DamlContextualizedErrorLoggerSpec extends AnyFlatSpec with Matchers {
   it should "sort entries by keys and skip empty values" in {
     val contextMap = Map("c" -> "C", "a" -> "A", "b" -> "B", "empty value" -> "")
 
-    val actual = ErrorCode.formatContextAsString(contextMap)
+    val actual =
+      DamlContextualizedErrorLogger.forTesting(getClass).formatContextAsString(contextMap)
```
Comment (skisel-da) [inline ledger/error/src/test/suite/scala/com/daml/error/DamlContextualizedErrorLoggerSpec.scala:17]: would be nice to align the spec name (`DamlContextualizedErrorLoggerSpec`) with the name of the class it is testing (i.e. `ContextualizedErrorLogger`), though I am not sure this method should belong then to that trait and be actually in `DamlContextualizedErrorLogger` object

File: ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala [hunk 3]
```diff
@@ -20,6 +19,18 @@ trait ContextualizedErrorLogger {
...
+    contextMap
+      .filter(_._2.nonEmpty)
+      .toSeq
+      .sortBy(_._1)
+      .map { case (k, v) =>
+        s"$k=$v"
+      }
+      .mkString(", ")
+  }
```
Comment (skisel-da) [inline ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala:33]: I am not sure, but how about this:
```
    contextMap
      .toSeq
      .sortBy(_._1)
      .collect {
        case (key, value) if value.nonEmpty => s"$key=$value"
      }
      .mkString(", ")
```

**Review by skisel-da — APPROVED** [review]

**Comment by pbatko-da** [comment]
\azp run

File: ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala (hunk 1 above)
Comment (pbatko-da) [reply ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala:22]: Ok, done

File: ledger/error/src/test/suite/scala/com/daml/error/DamlContextualizedErrorLoggerSpec.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/error/src/test/suite/scala/com/daml/error/DamlContextualizedErrorLoggerSpec.scala]: Ok, done

File: ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala (hunk 3 above)
Comment (pbatko-da) [reply ledger/error/src/main/scala/com/daml/error/ContextualizedErrorLogger.scala:31]: I just moved this function and would like to avoid changing it.

**Review by skisel-da — APPROVED** [review]

**Review by kamil-da — APPROVED** [review]

**Pull request merged**